Practice 1

StanislavMolchanovStudentKarazinUa opened the pull request.

StanislavMolchanovStudentKarazinUa committed:
Practice 1

StanislavMolchanovStudentKarazinUa committed:
Update practice1.cpp: Code redaction#001

Wrong use of delete function.
Changed delete(footbolist); to delete[] footbolist;

StanislavMolchanovStudentKarazinUa committed:
Update practice1.cpp: Code redaction#002

Changed function BestForward()
Reason: invalid line: footbolist++;

StanislavMolchanovStudentKarazinUa committed:
Report Extra: Input/Output cases

StanislavMolchanovStudentKarazinUa committed:
Update practice1.cpp: Code redaction#003

Now the function BestForward() compares not whole parts, but decimal fractions

mariiaShvedchenko commented on practice1.cpp:
```diff
@@ -0,0 +1,178 @@
+/////////////////////////////////////////////////////////////////////
+struct Footbolist {
+    string surname = "";
+    string role = "";
```
Better to do role an Enum.

mariiaShvedchenko commented on practice1.cpp:
```diff
@@ -0,0 +1,178 @@
+    Footbolist footbolist;
+    cout << "Footbolist's surname: ";
+    cin >> (&footbolist)->surname;
+    cout << "Footbolist's role: ";
+    cin >> (&footbolist)->role;
```
Here you do not have any check if the role of the football player is correct. Do "role" an enum or do checker for role to be correct

mariiaShvedchenko commented on practice1.cpp:
```diff
@@ -0,0 +1,178 @@
+        else {
+            int i;
+            Footbolist* newfootbolist = new Footbolist[(*n) - 1];
+            for (i = 0; i < k; i++) {
+                *(newfootbolist + i) = *(footbolist + i);
+            }
+            for (i = k + 1; i < *n; i++) {
+                *(newfootbolist + i - 1) = *(footbolist + i);
+            }
+            (*n)--;
```
If you already form a new array, why just not have an one for with one if statement there for skip element of array number k

mariiaShvedchenko commented on practice1.cpp:
```diff
@@ -0,0 +1,178 @@
+        (&footbolist1)->goal_count == (&footbolist2)->goal_count;
+}
+int FindFootbolist(Footbolist* footbolist, int n, Footbolist footbolist0) {
+    for (int i = 0; i < n; i++) if (SameFootbolist(footbolist[i], footbolist0)) return i;
+    return -1;
```
```suggestion
    for (int i = 0; i < n; i++) {
        if (SameFootbolist(footbolist[i], footbolist0)) {
            return i;
        }
     }
    return -1;

```

mariiaShvedchenko reviewed (commented):
so, do some changes and commit them. In general good work, keep going!)

StanislavMolchanovStudentKarazinUa committed:
Update practice1.cpp: Code redaction#004

Completed requests:
Better to do role an Enum.
Do "role" an enum or do checker for role to be correct

StanislavMolchanovStudentKarazinUa committed:
Update practice1.cpp: Code redaction#005

Completed request:
If you already form a new array, why just not have an one for with one if statement there for skip element of array number k

StanislavMolchanovStudentKarazinUa committed:
Update practice1.cpp: Code redaction#006

Completed request: Code redaction of FindFootbolist() function
